app.py

```diff
@@ -159,6 +159,9 @@ def main(
     subdir: Optional[str],
 ):
     """MI-Scheduler entrypoint."""
+    if subdir is None:
```

Comment:
I think you also need to adjust `subdir: Optional[str] = None` -> `subdir: str = ""`, because it can never be None, but rather empty string if it is part of a Path always https://github.com/thoth-station/mi-scheduler/blob/3d1b00836c2fcb71f173430485662f7ec0e84cdb/app.py#L52

Reply:
Well, it _can_ be None, if user does not want specify it.

By `if subdir is None` guard it is secured any case of `subdir` being `None` (if not specified or specified as the `None` - look example below).
Because of that, making the `subdir` parameter by default `None` seems to me kind of redundant then, because you can still do
```
>>> def foo(x = ""):
...     print(x)
... 
>>> foo(None)
None
```
which is not secured in this case. The guard on line 162 catches all of this.

Reply:
mmm 

```
print(type(""))
print(type(None))

<class 'str'>
<class 'NoneType'>
```

they are not same type and with the guard, you are never passing None, subdir: Optional[str] = None -> subdir: str = "", and if you pass None the path will be broken.

Reply:
> they are not same type and with the guard, you are never passing None,

yes, that should be the guard's purpose, right?
if the `subdir` is "", the it is alright (`Path` handles that) and if the `subidr` is `None`, then it is catched.
The `subdir: str = ""` set up does not guarantee that the `subdir` will always be non-`None` (example in previous comment)

Or am I missing something?